pr_agent/git_providers/github_provider.py

```diff
@@ -611,51 +619,35 @@ def remove_reaction(self, issue_comment_id: int, reaction_id: str) -> bool:
     def _parse_pr_url(self, pr_url: str) -> Tuple[str, int]:
         parsed_url = urlparse(pr_url)
 
-        path_parts = parsed_url.path.strip('/').split('/')
-        if self.base_domain in parsed_url.netloc:
-            if len(path_parts) < 5 or path_parts[3] != 'pulls':
-                raise ValueError("The provided URL does not appear to be a GitHub PR URL")
-            repo_name = '/'.join(path_parts[1:3])
+        pattern = r'/repos/([^/]+/[^/]+)/pulls?/(\d+)'
+        match = re.match(pattern, parsed_url.path)
+
+        if match:
+            repo_name = match.group(1)
             try:
-                pr_number = int(path_parts[4])
+                pr_number = int(match.group(2))
             except ValueError as e:
                 raise ValueError("Unable to convert PR number to integer") from e
             return repo_name, pr_number
-
-        if len(path_parts) < 4 or path_parts[2] != 'pull':
+        else:
             raise ValueError("The provided URL does not appear to be a GitHub PR URL")
 
-        repo_name = '/'.join(path_parts[:2])
-        try:
-            pr_number = int(path_parts[3])
-        except ValueError as e:
-            raise ValueError("Unable to convert PR number to integer") from e
-
-        return repo_name, pr_number
-
     def _parse_issue_url(self, issue_url: str) -> Tuple[str, int]:
         parsed_url = urlparse(issue_url)
-        path_parts = parsed_url.path.strip('/').split('/')
-        if self.base_domain in parsed_url.netloc:
-            if len(path_parts) < 5 or path_parts[3] != 'issues':
-                raise ValueError("The provided URL does not appear to be a GitHub ISSUE URL")
-            repo_name = '/'.join(path_parts[1:3])
+        path = parsed_url.path
+        
+        pattern = r'/repos/([^/]+/[^/]+)/issues/(\d+)'
+        match = re.match(pattern, path)
+        
+        if match:
+            repo_name = match.group(1)
             try:
-                issue_number = int(path_parts[4])
+                issue_number = int(match.group(2))
             except ValueError as e:
                 raise ValueError("Unable to convert issue number to integer") from e
             return repo_name, issue_number
-
-        if len(path_parts) < 4 or path_parts[2] != 'issues':
-            raise ValueError("The provided URL does not appear to be a GitHub PR issue")
-
-        repo_name = '/'.join(path_parts[:2])
-        try:
-            issue_number = int(path_parts[3])
-        except ValueError as e:
-            raise ValueError("Unable to convert issue number to integer") from e
-
-        return repo_name, issue_number
+        else:
+            raise ValueError("The provided URL does not appear to be a GitHub issue URL")
 
```

Comment:
does the new code supports:
'https://github.com/Codium-ai/pr-agent/issues/1268'
?